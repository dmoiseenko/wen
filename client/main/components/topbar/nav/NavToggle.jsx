import React from 'react';
import PropTypes from 'prop-types';
import classnames from 'classnames';


export default function NavToggle({ openCloseMenu, isOpen }) {
  function handleClick() {
    openCloseMenu();
  }

  const className = classnames('nav-toggle', {
    'is-active': isOpen
  });

  return (
    <span className={className} onClick={handleClick}>
      <span />
      <span />
      <span />
    </span>
  );
}

NavToggle.propTypes = {
  openCloseMenu: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired
};
