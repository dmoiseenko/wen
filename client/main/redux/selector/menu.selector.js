/* eslint-disable import/prefer-default-export */

export const isMenuOpenSelector = state => state.menu.isOpen;

/* eslint-enable */
