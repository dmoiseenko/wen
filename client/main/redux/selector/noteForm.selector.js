import { formValueSelector } from 'redux-form';

import { noteForm, textField } from '../../core/note.form.constant';


export const noteFromSelector = formValueSelector(noteForm);

export const noteTextSelector = state => noteFromSelector(state, textField);
