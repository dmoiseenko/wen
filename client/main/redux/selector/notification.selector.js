export const notifications = state => state.notifications; // eslint-disable-line
