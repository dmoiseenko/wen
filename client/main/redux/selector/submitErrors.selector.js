const validateInput = (state, formName, fieldName) => {
  if (!formName) {
    throw new Error('formName should be defined');
  }

  if (!state) {
    throw new Error('state should be defined');
  }

  if (!fieldName) {
    throw new Error('fieldName should be defined');
  }
};

const checkReduxFormSetup = (state) => {
  if (!state.form) {
    throw new Error('redux-form is not initialized');
  }
};

export default (state, formName, fieldName) => {
  validateInput(state, formName, fieldName);

  checkReduxFormSetup(state);

  if (!state.form[formName]) {
    return '';
  }

  if (!state.form[formName].submitErrors) {
    return '';
  }

  return state.form[formName].submitErrors[fieldName];
};
