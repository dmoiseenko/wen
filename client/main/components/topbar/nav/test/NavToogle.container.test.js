import * as container from '../NavToggle.container';
import { openCloseMenu } from '../../../../redux/modules/menu.module';
import * as selector from '../../../../redux/selector/menu.selector';


jest.mock('../../../../redux/selector/menu.selector', () => ({
  isMenuOpenSelector: jest.fn().mockReturnValue('isOpen')
}));

it('should map state to props correctly', () => {
  const actual = container.mapStateToProps('state');

  expect(actual).toMatchSnapshot();
  expect(selector.isMenuOpenSelector).toHaveBeenCalledWith('state');
});

it('should map dispatch to props correctly', () => {
  expect(container.mapDispatchToProps).toEqual({
    openCloseMenu
  });
});
