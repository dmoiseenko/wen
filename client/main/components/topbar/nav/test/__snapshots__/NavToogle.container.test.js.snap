// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should map state to props correctly 1`] = `
Object {
  "isOpen": "isOpen",
}
`;
